File: LeetCode/0038-count-and-say/0038-count-and-say.js
/**
 * @param {number} n
 * @return {string}
 */
var countAndSay = function(n) {
    var say = "1";
    for (let number = 2; number <= n; number++) {
        var count = 1;
        var result = "";
        for (let i = 0; i < say.length-1; i++) {
            if (say[i] === say[i+1]) count++;
            else {
                result += count.toString()+say[i];
                count = 1;
            }
        }
        result += count.toString()+say[say.length-1];
        say = result;
    }
    return say;
};

File: LeetCode/0038-count-and-say/0038-count-and-say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        say = ["", "1"]

        for i in range(2, n+1):
            string = say[i-1]
            prev = string[0]
            count = 1
            result = ""
            for j in range(1, len(string)):
                if prev == string[j]:
                    count += 1
                else:
                    result += str(count)+prev
                    count = 1
                prev = string[j]

            if count != 0:
                result += str(count)+prev
            
            say.append(result)

        print(say)
        return say[n]